client/my-sites/posts/index.js

```diff
@@ -1,16 +1,14 @@
 import page from '@automattic/calypso-router';
-import { makeLayout, render as clientRender, redirectIfDuplicatedView } from 'calypso/controller';
+import { redirectIfDuplicatedView, makeLayout, render as clientRender } from 'calypso/controller';
 import { getSiteFragment } from 'calypso/lib/route';
-import { navigation, siteSelection } from 'calypso/my-sites/controller';
-import postsController from './controller';
+import { siteSelection, sites } from 'calypso/my-sites/controller';
 
 export default function () {
 	page(
-		'/posts/:author(my)?/:status(published|drafts|scheduled|trashed)?/:domain?',
+		'/posts/:author?/:status(published|drafts|scheduled|trashed)?/:domain?',
 		siteSelection,
 		redirectIfDuplicatedView( 'edit.php' ),
-		navigation,
-		postsController.posts,
+		sites,
 		makeLayout,
 		clientRender
```

Comment:
We should remove all these middlewares; the code will always stop at `redirectIfDuplicatedView`.